src/engine/CMakeLists.txt

```diff
@@ -9,14 +9,41 @@ file(GLOB_RECURSE ENGINE_HEADER
         "*.hpp"
         )
 
+# ----------------------------------------------------
+# lib for editor
+
+add_library(leieditor_lib
+        ${ENGINE_HEADER}
+        ${ENGINE_SOURCE}
+        )
+
+add_dependencies (leieditor_lib glad)
+
+target_include_directories(leieditor_lib PUBLIC
+        .
+        ${GLFW_INCLUDE}
+        ${GLAD_INCLUDE}
+        ${IMGUI_INCLUDE}
+        ${SPDLOG_INCLUDE}
+        ${BULLET_INCLUDE}
+        ${MINIAUDIO_INCLUDE}
+        ${STB_INCLUDE}
+        ${ASSIMP_INCLUDE}
+        ${GLM_INCLUDE}
+        )
+
+target_compile_definitions(leieditor_lib PRIVATE -DEDITOR)
```

Comment:
> How is the EDITOR flag being defined when we make a Lei_Editor build?

@kevinsadi Lei_Editor now uses a separate library, compiled with the `EDITOR` flag

Reply:
ah makes sense